docs/decisions/0019-scaling-styles-with-design-tokens.rst

```diff
@@ -0,0 +1,261 @@
+
+Bootstrap 4 supports theming of visual styles through SCSS variables. As a result, Paragon's current theming system also relies overriding SCSS variables, both from core Bootstrap and custom Paragon variables.
+
+While Paragon's SCSS theming system has largely supported Open edX's transition from building legagy UIs in Django to React micro-frontends, we have identified several challenges that hinder its continued scalability and use in micro-frontends throughout Open edX.
+
+Challenges with current styles architecture
+-------------------------------------------
+
+* **On theme changes, all its consuming applications must be upgraded, re-built, and re-deployed.**
+
+  * Because Paragon currently heavily relies on SCSS variables for its theme, when any SCSS variable(s) are changed, every application consuming that theme needs to upgrade to the new version, be re-built, and get re-deployed.
+  
+    * With applications needing to be rebuild/redeployed, it's a significant effort to upgrade all applications to the new theme, especially if trying to do all applications at once.
+  
+  * With runtime theming via CSS variables, when a change is made to the theme, only the theme itself should need to get re-built, and the styles should automatically get applied to all consuming applications without needing to explicitly upgrade, re-build, and re-deploy those applications.
+
+* **Paragon styles are not platform agnostic nor a single source-of-truth.**
+
+  * The styles associated with Paragon themes are largely only accessible in environments that support SCSS as pretty much all Paragon's style properties are baked into SCSS variables.
+  * Ideally, the style properties would be defined in a platform agnostic way such that they can be transformed for various platforms, like iOS and Android apps or even a Figma integration.
+  
+    * *Note: these additional platform support ideas is more of a longer-term vision for how we could extend the design tokens work further in the future.*
+
+* **Supporting both system-wide themes and organizational themes.**
+
+  * The primary theming use case for Paragon is largely around system-wide theming, where all applications in the Open edX ecosystem share the same theme.
+  * However, there are use cases for organizational themes, too (i.e., updating the colors for specific partners/organizations, enterprise customers, etc.).
+  
+    * This is not well supported today and largely requires overriding CSS classes from Paragon rather than the desired approach of overriding underlying CSS variable(s). This an anti-pattern as Paragon class names should really be considered internal implementation details of Paragon components, and not used by consumers directly.
```

Comment:
It'd be nice to have an example of organizational theming being done the "anti-pattern way" vs the "design token way" to demonstrate this.

I'm not sure where exactly the specific example could fit into the ADR, but I think this is an incredible illustration of *why* design tokens/css variables are such a great approach here

Reply:
@brian-smith-tcril I'll give a go at including a brief example of the Enterprise MFEs relying on overriding class names where it would instead be great for those Enterprise MFEs to rely on the design tokens tooling / CSS variables instead for a more complete/holistic organization theming solution 👍

Let me know what you think of the following verbiage I just added to this paragraph:

> For example, the Enterprise MFEs within Open edX (e.g., frontend-app-learner-portal-enterprise) inject ``<style>`` tag in the HTML document with CSS class name overrides such as ``.btn-primary { background-color: $enterpriseCustomerPrimaryColor }``. Ideally, similar use cases could instead override a CSS variable such as ``--pgn-color-btn-primary-bg`` instead such that it gets re-used throughout all Paragon styles, not just ``.btn-primary``.